### PR TITLE
Docs: Add supported options for rewrite_data_files and rewrite_position_delete_files

### PR DESCRIPTION
**Pull request by manuzhang**
Currently, we point users to various Java docs, which don't contain much helpful information, for supported options in these actions. This PR assembles all the option names, values and descriptions in one table for these actions respectively.

### CODE REVIEW

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -333,9 +349,6 @@ Rewrite manifests for a table to optimize scan planning.
 
 Data files in manifests are sorted by fields in the partition spec. This procedure runs in parallel using a Spark job.
 
-See the [`RewriteManifests` Javadoc](../../../javadoc/{{% icebergVersion %}}/org/apache/iceberg/actions/RewriteManifests.html)
```
Comment (manuzhang) [inline docs/spark-procedures.md:336]: `rewrite_manifests` don't support options.

File: docs/spark-procedures.md [hunk 2]
```diff
@@ -283,11 +283,27 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
 | `options`     | ️   | map<string, string> | Options to be used for actions|
 | `where`       | ️   | string | predicate as a string used for filtering the files. Note that all files that may contain data matching the filter will be selected for rewriting|
 
+#### Options
+
+| Name | Default Value | Description |
+|------|---------------|-------------|
+| `max-concurrent-file-group-rewrites` | 5 | Maximum number of file groups to be simultaneously rewritten |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:290]: This is great, wondering, can we take the text directly from the javadoc?  Looks like that has more information.

File: docs/spark-procedures.md [hunk 3]
```diff
@@ -283,11 +283,27 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
...
+| `target-file-size-bytes` | default value of `write.target-file-size-bytes` from [table properties](../configuration/#write-properties) | Target output file size |
+| `min-file-size-bytes` | 75% of target file size | Files under this threshold will be considered for rewriting regardless of any other criteria |
+| `max-file-size-bytes` | 180% of target file size | Files with sizes above this threshold will be considered for rewriting regardless of any other criteria |
+| `min-input-files` | 5 | Any file group exceeding this number of files will be rewritten regardless of other criteria |
+| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
+| `max-file-group-size-bytes` | 107374182400 | Largest amount of data that should be rewritten in a single file group |
+| `delete-file-threshold` | 2147483647 | Minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting |
+| `compression-factor` | 1.0 | Let user adjust for file size used for estimating actual output data size (used with sort strategy) |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:302]: Do you think it is easier to split out strategy-specific ones into separate table?

**Review by szehon-ho — COMMENTED** [review]
Had a similar thought at some point, thanks for tackling this.

File: docs/spark-procedures.md (hunk 2 above)
Comment (manuzhang) [reply docs/spark-procedures.md:290]: Do you mean copy all the javadoc comments? Will that be too long? I'm trying to put a summary here.

File: docs/spark-procedures.md (hunk 3 above)
Comment (manuzhang) [reply docs/spark-procedures.md:302]: Currently, only sort-order has two separate configs and zorder has one. Do you think it's worth another table?

File: docs/spark-procedures.md [hunk 4]
```diff
@@ -283,11 +283,27 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
...
+| `shuffle-partitions-per-file` | 1 | Number of shuffle partitions to use for each output file (used with sort strategy) |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:303]: I think we add this sentence back:
```
Iceberg will use a custom coalesce operation to stitch these sorted partitions back together into a single sorted file.
```

File: docs/spark-procedures.md (hunk 3 above)
Comment (szehon-ho) [reply docs/spark-procedures.md:302]: Yea I think we can add sub-sections like:

- General options
- Options used when sort_order is set
- Options used when strategy is z_order

Please check and correct

File: docs/spark-procedures.md (hunk 2 above)
Comment (szehon-ho) [reply docs/spark-procedures.md:290]: Yea took a second look, you are right, some of them are too long.  

But now we are removing references to javadoc, I suggest to still add back more context, I added suggestion to individual options.

**Comment by manuzhang** [comment]
@szehon-ho Thanks for all good suggestions. Please take another look.

File: docs/spark-procedures.md [hunk 5]
```diff
@@ -283,11 +283,27 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
...
+| `use-starting-sequence-number` | true | Use the sequence number of the snapshot at compaction start time instead of that of the newly produced snapshot |
+| `rewrite-job-order` | none | Force the rewrite job order based on the value (one of bytes-asc, bytes-desc, files-asc, files-desc, none) |
+| `target-file-size-bytes` | default value of `write.target-file-size-bytes` from [table properties](../configuration/#write-properties) | Target output file size |
+| `min-file-size-bytes` | 75% of target file size | Files under this threshold will be considered for rewriting regardless of any other criteria |
+| `max-file-size-bytes` | 180% of target file size | Files with sizes above this threshold will be considered for rewriting regardless of any other criteria |
+| `min-input-files` | 5 | Any file group exceeding this number of files will be rewritten regardless of other criteria |
+| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
+| `max-file-group-size-bytes` | 107374182400 | Largest amount of data that should be rewritten in a single file group |
```
[the remark this answers is not in the document]
Comment (manuzhang) [reply docs/spark-procedures.md]: Added. It's 100GB actually ;)

File: docs/spark-procedures.md [hunk 6]
```diff
@@ -283,11 +283,37 @@ Iceberg can compact data files in parallel using Spark with the `rewriteDataFile
...
+| `max-file-size-bytes` | 180% of target file size | Files with sizes above this threshold will be considered for rewriting regardless of any other criteria |
+| `min-input-files` | 5 | Any file group exceeding this number of files will be rewritten regardless of other criteria |
+| `rewrite-all` | false | Force rewriting of all provided files overriding other options |
+| `max-file-group-size-bytes` | 107374182400 (100GB) | Largest amount of data that should be rewritten in a single file group. The entire rewrite operation is broken down into pieces based on partitioning and within partitions based on size into file-groups.  This helps with breaking down the rewriting of very large partitions which may not be rewritable otherwise due to the resource constraints of the cluster. |
+| `delete-file-threshold` | 2147483647 | Minimum number of deletes that needs to be associated with a data file for it to be considered for rewriting |
+| `max-output-size` | 2147483647 | Amount of bytes interleaved in the ZOrder algorithm (used with zorder) |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:303]: Should we move it to zorder?

File: docs/spark-procedures.md [hunk 7]
```diff
@@ -379,11 +403,23 @@ Iceberg can rewrite position delete files, which serves two purposes:
...
+
+| Name | Default Value | Description |
+|------|---------------|-------------|
+| `max-concurrent-file-group-rewrites` | 5 | Maximum number of file groups to be simultaneously rewritten |
+| `partial-progress.enabled` | false | Enable committing groups of files prior to the entire rewrite completing |
+| `partial-progress.max-commits` | 10 | Maximum amount of commits that this rewrite is allowed to produce if partial progress is enabled |
+| `rewrite-job-order` | none | Force the rewrite job order based on the value. <ul><li>If rewrite-job-order=bytes-asc, then rewrite the smallest job groups first.</li><li>If rewrite-job-order=bytes-desc, then rewrite the largest job groups first.</li><li>If rewrite-job-order=files-asc, then rewrite the job groups with the least files first.</li><li>If rewrite-job-order=files-desc, then rewrite the job groups with the most files first.</li><li>If rewrite-job-order=none, then rewrite job groups in the order they were planned (no specific ordering).</li></ul> |
+| `target-file-size-bytes` | 67108864 (default value of `write.delete.target-file-size-bytes` from [table properties](../configuration/#write-properties)) | Target output file size |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:416]: Sorry just noticed a small discrepancy, we mention the number here but not in the same config of rewrite_data_files where we just mention link, can we make make it consistent (no strong preference either way)?

**Review by szehon-ho — APPROVED** [review]
Looks good to me, one small remaining nit.

File: docs/spark-procedures.md [hunk 8]
```diff
@@ -379,11 +403,23 @@ Iceberg can rewrite position delete files, which serves two purposes:
...
+| `partial-progress.enabled` | false | Enable committing groups of files prior to the entire rewrite completing |
+| `partial-progress.max-commits` | 10 | Maximum amount of commits that this rewrite is allowed to produce if partial progress is enabled |
+| `rewrite-job-order` | none | Force the rewrite job order based on the value. <ul><li>If rewrite-job-order=bytes-asc, then rewrite the smallest job groups first.</li><li>If rewrite-job-order=bytes-desc, then rewrite the largest job groups first.</li><li>If rewrite-job-order=files-asc, then rewrite the job groups with the least files first.</li><li>If rewrite-job-order=files-desc, then rewrite the job groups with the most files first.</li><li>If rewrite-job-order=none, then rewrite job groups in the order they were planned (no specific ordering).</li></ul> |
+| `target-file-size-bytes` | 67108864 (64MB, default value of `write.delete.target-file-size-bytes` from [table properties](../configuration/#write-properties)) | Target output file size |
+| `min-file-size-bytes` | 75% of target file size | Files under this threshold will be considered for rewriting regardless of any other criteria |
```
Comment (szehon-ho) [inline docs/spark-procedures.md:417]: Nit: somehow 'with sizes' got lost in the paste, unlike for max-file-size-bytes

Comment (manuzhang) [reply docs/spark-procedures.md:417]: I will keep this in mind and fix it next time.

**Comment by manuzhang** [comment]
@szehon-ho thanks for proofreading and nice suggestions.

**Pull request merged**